src/test/java/org/candlepin/config/FakeConfig.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.config;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class FakeConfig implements Configuration {
+    private final Map<String, String> config;
+
+    public FakeConfig() {
+        this(new HashMap<>());
+    }
+
+    public FakeConfig(Map<String, String> config) {
+        this.config = Objects.requireNonNull(config);
```

Comment:
Good point. Updated to copy the map. Changing the map directly can get confusing. This way it can be controlled by either pre-initialized map or config mutators.